### PR TITLE
[fix] Function percentile input null return null

### LINKED ISSUE #8236: [Bug] Function percentile input null return 0
### Search before asking

- [X] I had searched in the [issues](https://github.com/apache/incubator-doris/issues?q=is%3Aissue) and found no similar issues.

### Version

current master

### What's Wrong?

select percentile(NULL,0.3) from table1;   -----> return 0;

this function in Hive will be return NULL

### What You Expected?

**input NULL return NULL**

### How to Reproduce?

_No response_

### Anything Else?

_No response_

### Are you willing to submit PR?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by zhangstar333**
# Proposed changes

Issue Number: close #8236 

## Problem Summary:

Describe the overview of changes.

## Checklist(Required)

1. Does it affect the original behavior: (Yes/No/I Don't know)
2. Has unit tests been added: (Yes/No/No Need)
3. Has document been added or modified: (Yes/No/No Need)
4. Does it need to update dependencies: (Yes/No)
5. Are there any changes that cannot be rolled back: (Yes/No)

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Review by HappenLee — DISMISSED** [review]
LGTM

File: be/src/vec/functions/function_timestamp.cpp [hunk 1]
```diff
@@ -52,8 +52,60 @@ struct StrToDate {
...
+                ts_val.to_datetime();
+            } else {
+                ts_val.cast_to_date();
+            }
+        }
+    }
+};
+
+template <typename Impl>
+class FunctionStrToDateOrDateTime : public IFunction {
```
Comment (HappenLee) [inline be/src/vec/functions/function_timestamp.cpp:66]: should change other function of `FunctionBinaryStringOperateToNullType` and add `FunctionContext`

**Review by HappenLee — COMMENTED** [review]
some problem

File: be/src/vec/functions/function_timestamp.cpp (hunk 1 above)
Comment (zhangstar333) [reply be/src/vec/functions/function_timestamp.cpp]: done

**Review by HappenLee — DISMISSED** [review]
LGTM

**Review by HappenLee — APPROVED** [review]
LGTM

**Pull request merged**